docs/content/manual/_incoming/2342_Instance_metadata_variables_are_not_expanded.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Instance metadata variables are not expanded
+category: UI
+tags: VM, p1, functional
+---
+Ref: https://github.com/harvester/harvester/issues/2342
+
+![image](https://user-images.githubusercontent.com/5169694/177121301-f30bf8ec-0a70-4549-b11b-895161ee30ad.png)
+
+
+### Verify Steps:
+1. Install Harvester with any nodes
+1. Create Image for VM creation
+1. Create VM with following _CloudConfig_
+```yaml
+## template: jinja
+#cloud-config
+package_update: true
+password: password
+chpasswd: { expire: False }
+sshpwauth: True
+write_files:
+  - content: |
+      #!/bin/bash
+      vmName=$1
+      echo "VM Name is: $vmName" > /home/cloudinitscript.log
+    path: /home/exec_initscript.sh
+    permissions: '0755'
+runcmd:
+  - - systemctl
+    - enable
+    - --now
+    - qemu-guest-agent.service
+  - - echo
+    - "{{ ds.meta_data.local_hostname }}"
+  - - /home/exec_initscript.sh
+    - "{{ ds.meta_data.local_hostname }}"
```

Comment:
This is really cool :+1: :smile: !  I didn't know that there were a bunch of built-in options for jinja templating in cloud-config!  I had no clue that we could access elements from `ds` - do we have `ds` attributes documented anywhere in any docs?  Or does everything from `ds` come from KubeVirt? - just curious :smile:

Reply:
FYR: https://cloudinit.readthedocs.io/en/latest/topics/instancedata.html